content/id-datatype/index.md

```diff
@@ -0,0 +1,94 @@
+---
+rfc:
+start_date: 2018-08-02
+pr: openregister/registers-rfcs#22
+status: draft
+---
+
+# ID datatype
+
+## Summary
+
+This RFC proposes a new datatype to constrain and govern identifiers for
+records.
+
+
+## Motivation
+
+Currently, identifiers (i.e. keys) are defined as of any datatype available.
+Usage shows only strings have been used thus far.
+
+Allowing any datatype imposes a hard task on the user trying to validate
+what to expect in there. Other questions arise when you analyise how each
+datatype is expected to behave in all contexts identifiers are used.
+
+Questions like “What are the rules for an identifier when used as part of a
+CURIE?”, “What does it mean to have a String (full UTF-8 range)?”.
+
+Notice that [issue 18](https://github.com/openregister/registers-rfcs/issues/18)
+has a discussion related to this RFC.
+
+This RFC aims to improve the situation by proposing a new ID datatype
+constrained enough to make validation viable, that works well with CURIEs and
+overall that guides users without preventing valid use cases.
+
+
+## Explanation
+
+The new ID datatype is a subset of the String datatype. It helps keeping
+backwards compatibility and builds on top of how Registers are used in
+reality.
+
+Evidence says that identifiers are composed primarily by alphanumeric ASCII
+characters with a few other printable characters:
+
+* `-` (U+002D)
+* `_` (U+005F)
+* `.` (U+002E)
+* `/` (U+002F)
+
+Note that `/` has meaning in URLs and hence it has meaning in CURIEs. This
+means that a CURIE must always percent-encode this character as `%2F`.
+
+The proposal then is to not allow more characters than the ones that are
+currently used:
+
+```abnf
+id = (ALPHA / DIGIT) *(ALPHA / DIGIT / restricted)
+restricted = delimiters / legacy
+delimiters = "-" / "_" / "."
+legacy = "%2F"
```

Comment:
This implies '/' is always percent encoded in IDs but the text above says it's only percent encoded when used as a CURIE.

Reply:
Very true, I'll amend this, thanks